### ISSUE TITLE
"Chunker.app" is damaged and can't be opened.

### ISSUE THREAD

**Issue by ethankrech** [HiveGamesOSS/Chunker#157]
### Chunker Version

1.4.1-arm64-mac

### Platform / Operating System

MacOS (latest)

### Issue Description

Title. On latest MacOS update, 15.1.1.

<img width="257" alt="Screenshot 2024-11-27 at 10 38 07 PM" src="https://github.com/user-attachments/assets/3a3d2889-788a-4611-8f28-b1d9e9a46252">

**Comment by FormallyMyles** [comment]
We currently don't sign the mac version which leads to this, I'll see where we're at with this with the team.

You can work around this with the .zip version by removing the flag which tells mac that it was downloaded `xattr -c Chunker.app`. Alternatively, you can build the software yourself and it would also allow it to run. 

If you don't need the GUI version, the `chunker-1.4.1-cli.jar` would also work via `java -jar chunker-1.4.1-cli.jar -i my_world/ -o output/ -f JAVA_1_21` (with Java installed).

**Comment by cianiandreadev** [comment]
Same issue here, but seems that the `xattr -c Chunker.app` worked indeed. 
Thanks

**Comment by ethankrech** [comment]
Occurs on 1.4.3 as well

**Comment by ethankrech** [comment]
Do you know when this will be resolved?

**Comment by FormallyMyles** [comment]
> Do you know when this will be resolved?

Unfortunately, the process of getting this setup with Apple is taking longer than we expected.

**Comment by Oliver81201** [comment]
> Same issue here, but seems that the `xattr -c Chunker.app` worked indeed. Thanks

What does this mean? Do I put this into terminal and run it? Sorry I'm new here

**Comment by FormallyMyles** [comment]
> > Same issue here, but seems that the `xattr -c Chunker.app` worked indeed. Thanks
> 
> What does this mean? Do I put this into terminal and run it? Sorry I'm new here

Yeah you need to run it in Terminal in the same folder as where you downloaded Chunker.app (changing Chunker.app to the full name of the file).

**Comment by hendrixfur** [comment]
hey, I am running into the same issue, Having a little trouble. so i would just change the zip file to remove that?

**Comment by FormallyMyles** [comment]
> hey, I am running into the same issue, Having a little trouble. so i would just change the zip file to remove that?

I believe you can do that, or unzip and apply it to the .app file.

**Comment by jlou1710** [comment]
I am on struggle street with this and am not as tech literate as everyone else, I figured out what Terminal is and how to open it but the command won't work, it says no such file and I thought that renaming the app in Finder to Chunker.app would help but it won't allow me to do this I'm at a loss I don't want to have to leave behind the world I've played for over 600 in-game days

**Comment by soymilkis** [comment]
hi! i'm just a girl trying to covert my bedrock world to java, and i was following these steps but i got a little confused! if u aren't code-savvy like me, here's how i did it. _for context, i'm using my macbook air MacOs x64, NOT ARM !_

go to the official releases (https://github.com/HiveGamesOSS/Chunker/releases) and download the mac.zip (i downloaded the amd64-mac.zip)

open the zip, then open terminal and paste in cd ~/Downloads/ and then from there, you should be able to paste xattr -c Chunker.app and then open chunker as normally!

i understand this is what the comments had already said, but i just decided to leave a step by step in case anyone is silly like me !

**Comment by legocub30** [comment]
Still Happens on 1.14.0-arm64-mac.dmg, MacOS 26.0.1

**Comment by KyaM8248541** [comment]
this still happens and its 2026...

**Comment by iosdevben** [comment]
As noted by @jcavalieri, in Terminal, go to `/Applications`, then run `xattr -c Chunker.app`.

Then you'll be able to double click on the Chunker app in Finder.

**Comment by JEMcats** [comment]
> go to `/Applications`, then run `xattr -c Chunker.app`.
> 
> Then you'll be able to double click on the Chunker app in Finder.

You can also use ```xattr -rd com.apple.quarantine Chunker.app```. A quick way to get the path to the app is to paste ```xattr -rd com.apple.quarantine ``` or ```xattr -c ``` into the terminal and then drag and drop the Chunker.app from Finder into the terminal, that will autofill the full path which works without depending on your current directory!
